url20b0f880a5_jgraph_legacy_jgraph5_tgz-pJ8-org_jgraph_example_HelloWorldJ8/src/com/mxgraph/layout/mxStackLayout.java

```diff
@@ -0,0 +1,330 @@
+	 */
+	protected boolean fill = false;
+
+	/**
+	 * If the parent should be resized to match the width/height of the
+	 * stack. Default is false.
+	 */
+	protected boolean resizeParent = false;
+
+	/**
+	 * Value at which a new column or row should be created. Default is 0.
+	 */
+	protected int wrap = 0;
+
+	/**
+	 * Constructs a new stack layout layout for the specified graph,
+	 * spacing, orientation and offset.
+	 */
+	public mxStackLayout(mxGraph graph)
+	{
+		this(graph, true);
+	}
+
+	/**
+	 * Constructs a new stack layout layout for the specified graph,
+	 * spacing, orientation and offset.
+	 */
+	public mxStackLayout(mxGraph graph, boolean horizontal)
+	{
+		this(graph, horizontal, 0);
+	}
+
+	/**
+	 * Constructs a new stack layout layout for the specified graph,
+	 * spacing, orientation and offset.
+	 */
+	public mxStackLayout(mxGraph graph, boolean horizontal, int spacing)
+	{
+		this(graph, horizontal, spacing, 0, 0, 0);
+	}
+
+	/**
+	 * Constructs a new stack layout layout for the specified graph,
+	 * spacing, orientation and offset.
+	 */
+	public mxStackLayout(mxGraph graph, boolean horizontal, int spacing,
+			int x0, int y0, int border)
+	{
+		super(graph);
+		this.horizontal = horizontal;
+		this.spacing = spacing;
+		this.x0 = x0;
+		this.y0 = y0;
+		this.border = border;
+	}
+	
+	/**
+	 * 
+	 */
+	public boolean isHorizontal()
+	{
+		return horizontal;
+	}
+
+	/*
+	 * (non-Javadoc)
+	 * @see com.mxgraph.layout.mxGraphLayout#move(java.lang.Object, double, double)
+	 */
+	public void moveCell(Object cell, double x, double y)
+	{
+		mxIGraphModel model = graph.getModel();
+		Object parent = model.getParent(cell);
+		boolean horizontal = isHorizontal();
+
+		if (cell instanceof mxICell && parent instanceof mxICell)
+		{
+			int i = 0;
+			double last = 0;
+			int childCount = model.getChildCount(parent);
+			double value = (horizontal) ? x : y;
+			mxCellState pstate = graph.getView().getState(parent);
+
+			if (pstate != null)
+			{
+				value -= (horizontal) ? pstate.getX() : pstate.getY();
+			}
+
+			for (i = 0; i < childCount; i++)
+			{
+				Object child = model.getChildAt(parent, i);
+
+				if (child != cell)
+				{
+					mxGeometry bounds = model.getGeometry(child);
+
+					if (bounds != null)
+					{
+						double tmp = (horizontal) ? bounds.getX()
+								+ bounds.getWidth() / 2 : bounds.getY()
+								+ bounds.getHeight() / 2;
+
+						if (last < value && tmp > value)
+						{
+							break;
+						}
+
+						last = tmp;
+					}
+				}
+			}
+
+			// Changes child order in parent
+			int idx = ((mxICell) parent).getIndex((mxICell) cell);
+			idx = Math.max(0, i - ((i > idx) ? 1 : 0));
+
+			model.add(parent, cell, idx);
+		}
+	}
+
+	/**
+	 * Hook for subclassers to return the container size.
+	 */
+	public mxRectangle getContainerSize()
+	{
+		return new mxRectangle();
+	}
+
+	/*
+	 * (non-Javadoc)
+	 * @see com.mxgraph.layout.mxIGraphLayout#execute(java.lang.Object)
+	 */
+	public void execute(Object parent)
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 32. By comparison, 99% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test. We recommend that you simplify this method or break it into multiple methods.